packages/salesforcedx-vscode-apex/src/index.ts

```diff
@@ -12,23 +12,84 @@ import {
   DEBUGGER_LINE_BREAKPOINTS
 } from './constants';
 import * as languageServer from './languageServer';
+import { ApexTestOutlineProvider } from './views/testOutline';
+
+export type ApexTestRequestInfo = {
+  methodName: string;
+  parent: string;
+  line: number;
+  file: string;
+};
 
 let languageClient: LanguageClient | undefined;
 let languageClientReady = false;
 
 export async function activate(context: vscode.ExtensionContext) {
   languageClient = await languageServer.createLanguageServer(context);
-  const handle = languageClient.start();
-  context.subscriptions.push(handle);
+  if (languageClient) {
+    const handle = languageClient.start();
+    context.subscriptions.push(handle);
+
+    languageClient.onReady().then(() => {
+      languageClientReady = true;
+    });
+  }
+
+  // Test View
+  const rootPath = vscode.workspace.rootPath || '';
```

Comment:
Use `vscode.workspace.workspaceFolders[0]` instead of rootPath since it's deprecated (https://github.com/Microsoft/vscode/wiki/Extension-Authoring:-Adopting-Multi-Root-Workspace-APIs)